File: ATC001/a.cpp
#include <iostream>
using namespace std;

int H, W;
char c[600][600];
int sy, sx;

bool dfs(int y, int x) {
    if ((y<0) || (y>=H) || (x<0) || (x>=W) || (c[y][x] == '#')) return false;
    else if (c[y][x] == 'g') return true;

    c[y][x] = '#';
    return (dfs(y-1, x) || dfs(y+1, x) || dfs(y, x-1) || dfs(y, x+1));
}

int main() {
    cin >> H >> W;
    for (int i=0; i<H; ++i) {
        for (int j=0; j<W; ++j) {
            cin >> c[i][j];
            if (c[i][j] == 's') {
                sy = i;
                sx = j;
            }
        }
    }
    
    if (dfs(sy, sx)) cout << "Yes" << endl;
    else cout << "No" << endl;

    return 0;
}

File: ATC001/a.py
H, W = map(int, input().split(" "))
c = []

for i in range(H):
    row = []
    for j,x in enumerate(input()):
        if x == "s":
            x0 = j
            y0 = i
        elif x == "g":
            x1 = j
            y1 = i
        row.append(x)
    c.append(row)

lst = []

def dfs(x, y):
    if x < 0 or x >= W or y < 0 or y >= H:
        return
    if (x, y) in lst:
        return
    if c[y][x] == "#":
        return
    elif c[y][x] == "g":
        lst.append((x, y))
        return
    else:
        lst.append((x, y))
        dfs(x+1, y)
        dfs(x-1, y)
        dfs(x, y+1)
        dfs(x, y-1)
    return

dfs(x0, y0)
if (x1, y1) in lst:
    print("Yes")
else:
    print("No")